### PR TITLE
FIAM FIS Integration 

### PR DESCRIPTION
**Pull request by JasonAHeron**

### CODE REVIEW

File: firebase-inappmessaging-display/src/main/java/com/google/firebase/inappmessaging/display/FirebaseInAppMessagingDisplay.java [hunk 1]
```diff
@@ -76,7 +76,7 @@
  * </ul>
  *
  * <p>To delete the Instance ID and the data associated with it, see {@link
- * com.google.firebase.iid.FirebaseInstanceId#deleteInstanceId}.
+ * FirebaseInstallationsApi#delete.
```
Comment (rlazo) [inline firebase-inappmessaging-display/src/main/java/com/google/firebase/inappmessaging/display/FirebaseInAppMessagingDisplay.java:79]: missing } after delete

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// limitations under the License.
+
+package com.google.firebase.inappmessaging.internal;
+
+import android.text.TextUtils;
+import com.google.firebase.installations.InstallationTokenResult;
+
+public class InstallationIdResult {
+  private String id;
+  private InstallationTokenResult token;
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java:22]: private final

File: firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.inappmessaging.internal;
+
+import android.text.TextUtils;
+import com.google.firebase.installations.InstallationTokenResult;
+
+public class InstallationIdResult {
```
Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java:20]: Do we need this to be `public`.. I think package private would work

Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java:20]: final

Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java:20]: docs on this class?

Comment (prakhar1989) [inline firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java:20]: Ideally we should be using https://github.com/google/auto/blob/master/value/userguide/index.md for these value objects. If its not too much work, consider adding using this lib

Comment (JasonAHeron) [reply firebase-inappmessaging/src/main/java/com/google/firebase/inappmessaging/internal/InstallationIdResult.java]: turns out we access this from two different packages for tests

**Pull request merged**